src/rpc/rpc_args.cpp

```diff
@@ -81,6 +90,34 @@ namespace cryptonote
         return boost::none;
       }
     }
+    if (!config.bind_ipv6_address.empty())
+    {
+      // allow square braces, but remove them here if present
+      if (config.bind_ipv6_address.find('[') != std::string::npos)
+      {
+        config.bind_ipv6_address = config.bind_ipv6_address.substr(1, config.bind_ipv6_address.size() - 2);
```

Comment:
That seems to remove any last char, regardless of whether ] or not.

Reply:
If the address was specified with an opening square brace at the beginning and not a closing square brace at the end then it was specified incorrectly.  I reasoned that it would be incredibly tedious to look for all of the possible permutations of incorrect address specification and give hints (e.g. "missing ] character after address") as there are quite a few ways of messing that up.  In this case if the ] is missing then when the code tries to use the address it will fail and message/error accordingly.